src/main/java/com/google/sps/storage/CaptionStorageManager.java

```diff
@@ -179,18 +181,77 @@ public boolean videoInDb(String videoID) throws CaptionStorageException {
 
     // return true if specified meta is the metadata for videoID
     public boolean metaInDb(String videoID, String meta) throws CaptionStorageException {
-        Entity data;
-        try {
-            data = getMetadata(videoID);
-        } catch (CaptionStorageException cse) {
-            throw cse;
-        }
+        Entity data = getMetadata(videoID);
         if (data.getKey().getName().equals(meta)) {
             return true;
-        } 
+        }
         return false;
     }
 
+    // delete a video & all its children (metadata & captions) from the database
+    public void deleteVideo(String videoID) throws CaptionStorageException {
+        Entity vidEnt;
+        if ((vidEnt = getVideo(videoID)) == null) {
+            throw new CaptionStorageException(Reason.NO_VIDEO_EXISTS, "Requested video does not exist");
```

Comment:
I would set the string as a static constant

Reply:
done